lib/methods/ifwapichecktask.cpp

```diff
@@ -269,18 +172,20 @@ static void DoIfwNetIo(
 
 		for (auto& pv : perfdata) {
 			if (!pv.IsString()) {
-				ReportIfwCheckResult(
-					checkable, cmdLine, cr,
+				cr->SetOutput(
 					"Got bad perfdata value " + JsonEncode(perfdata) + " from IfW API on host '"
-						+ psHost + "' port '" + psPort + "', expected an array of strings",
-					start, end
+						+ psHost + "' port '" + psPort + "', expected an array of strings"
 				);
 				return;
 			}
 		}
+
+		cr->SetPerformanceData(PluginUtility::SplitPerfdata(perfdata->Join(" ")));
 	}
 
-	ReportIfwCheckResult(checkable, cmdLine, cr, result->Get("checkresult"), start, end, exitcode, perfdata);
+	cr->SetExitStatus(exitcode);
+	cr->SetOutput(result->Get("checkresult"));
+	cr->SetState(static_cast<ServiceState>(exitcode.Get<double>()));
```

Comment:
Would be nicer if that casting happened at the same place where the corresponding validation happens.

Reply:
This would change the previous behaviour for the two error cases following exit code validation, i.e. previously the state would be set to unknown if invalid perfdata was provided, with your proposed change this would not be the case.

Reply:
The casting, not the `cr->SetState()` call. More like changing the type of `exitcode` (interesting how this conversion partially got hidden in the function call before).